lintro/utils/output/manager.py

```
@@ -56,27 +60,94 @@ def __init__(
+
+        Returns:
+            Path to the created run directory.
+        """
+        base_name = f"{DEFAULT_RUN_PREFIX}{timestamp}-{os.getpid()}"
+        run_dir = base_dir / base_name
+        suffix = 0
+
+        while True:
+            try:
+                run_dir.mkdir(parents=True, exist_ok=False)
+                return run_dir
```

Ruff (0.15.10)

[warning] Consider moving this statement to an `else` block

(TRY300)